src/modules/sub-items/sub.items.module.js

```diff
@@ -572,23 +750,46 @@ export default class SubItemsModule extends Component {
     }
 
     /**
-     * Renders load more button
+     * Renders viewing message
      *
-     * @method renderLoadMore
+     * @method renderViewingMessage
+     * @returns {JSX.Element}
+     */
+    renderViewingMessage() {
+        const { totalCount } = this.state;
+        const { activePage, pages } = this.state;
+        const pageData = pages[activePage];
+        const pageItems = pageData ? pageData.items : [];
+        const viewingCount = pageItems ? pageItems.length : 0;
```

Comment:
hmm `pageItems` is an array (empty or populated), so `pageItems.length` is enough?

Reply:
No 😄 `pageData` contains `{ isLoading: boolean, items: Array }` and items may be null, because in `setPageLoadingState` I set the pageData to: `{ isLoading: true, items: null }`.

I'm setting it to `null`, because I have this condition:
```js
const activePageData = pages[activePage];
const showSpinner = !activePageData || !activePageData.items;

if (showSpinner) {
     return this.renderSpinner();
}
```
in `renderListView`, because when we don't have subitems in given location we want to pass empty array to `SubItemsListComponent` (as before) and `<NoItemsComponent>` is rendered by them. What do you think? Maybe I should change it so that we would render `<NoItemsComponent>` in `sub.items.component.js` based on `state.totalCount`? It may be simpler.

Reply:
It seems like overcomplicated.

Correct me if I'm wrong:
1. `renderSpinner()` - renders spinner when items are loading.
2. You have flag `isLoading`
3. You render spinner when there are no items

Why not when `isLoading` is `true`?

Reply:
> Why not when isLoading is true?
Because on the first hit you don't have `pageData` for given `activePage` - it is set in `componentDidUpdate`, so on the first hit `pageData` is undefined so basically `pageData. isLoading == false` (because it is undefined) but you should treat it as `isLoading`.

Reply:
It feels like something is wrong in the architecture, we have flag `isLoading` but the flag doesn't answer the question if page is loading.

Reply:
We should definitely use this flag to display the loading spinner, unless we want to mess up our code. If the `pageData` is undefined, then you have to find the answer why is it set this way.
On first render we get all data for subitems from the backend, so no initial loading is required. You have to check how the code works and understand it. Adding additional code that duplicates existing solutions is not right way of development.

Reply:
> If the pageData is undefined, then you have to find the answer why is it set this way.

I think that my explanation wasn't clear enough, when saying this:
> Because on the first hit you don't have pageData for given activePage - it is set in componentDidUpdate, so on the first hit pageData is undefined so basically pageData. isLoading == false (because it is undefined) but you should treat it as isLoading.

by `on the first hit` I was talking about "first hit" after switching to a page not yet loaded - so I don't have data for it. I trigger loading in `componentDidUpdate`. :) 

So it's like:
1. switch to a new page (not loaded yet),
2. render triggers after state change (`pageData` not set),
3. detect in `componentDidUpdate` that activePage has not been loaded yet and trigger loading (set `isLoading` flag).

Reply:
Maybe you should set base, empty state for a page without data when trying to load it?

Reply:
I will try. :)